view/adminhtml/web/js/form/field/company-code.js

```diff
@@ -0,0 +1,79 @@
+define(['jquery', 'mage/url'], function (jQuery) {
```

Comment:
@LordZardeck - Please re-factor this JS file to be structured as a class, in line with any other Magento JS file that extends jQuery.widget and follows that pattern.  (e.g., use "$.widget(<name>, {" to create your object, make the functions object methods instead, make things like companyIdToCompanyCodeMap actual class properties, put the logic that's immediately executed in the "init" method, etc)